make/go/go.mk

```diff
@@ -20,6 +20,8 @@ GO_GET_PKGS ?=
 # Settable
 GO_MOD_VERSION ?= 1.21
 # Settable
+GO_MOD_TOOLCHAIN ?= 1.23.0
```

Comment:
We could set `+auto` to allow higher versions?

Reply:
Let's do this for now and then change it later if we want